source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/deploy/mdp/observations.py

```diff
@@ -0,0 +1,299 @@
+# Copyright (c) 2022-2025, The Isaac Lab Project Developers (https://github.com/isaac-sim/IsaacLab/blob/main/CONTRIBUTORS.md).
+# All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+"""Class-based observation terms for the gear assembly manipulation environment."""
+
+from __future__ import annotations
+
+import torch
+from typing import TYPE_CHECKING
+
+from isaaclab.assets import RigidObject
+from isaaclab.managers import ManagerTermBase, ObservationTermCfg, SceneEntityCfg
+from isaaclab.utils.math import combine_frame_transforms
+
+if TYPE_CHECKING:
+    from isaaclab.envs import ManagerBasedRLEnv
+
+    from .events import RandomizeGearType
+
+
+class GearShaftPosW(ManagerTermBase):
+    """Gear shaft position in world frame with offset applied.
+
+    This class-based term caches gear offset tensors and identity quaternions.
+    """
+
+    def __init__(self, cfg: ObservationTermCfg, env: ManagerBasedRLEnv):
+        """Initialize the gear shaft position observation term.
```

Comment:
Would be good to elaborate in the docstrings what the term args are. Please check the documentation here for examples: https://isaac-sim.github.io/IsaacLab/main/source/api/lab/isaaclab.envs.mdp.html#isaaclab.envs.mdp.observations.image_features